nym-api/src/node_describe_cache/mod.rs

```diff
@@ -194,51 +197,104 @@ async fn get_gateway_description(
         ip_packet_router,
         mixnet_websockets: websockets.into(),
         auxiliary_details,
+        role: data.role(),
     };
 
-    Ok((gateway.identity_key, description))
+    Ok((data.identity_key().clone(), description))
+}
+
+struct RefreshData {
+    host: String,
+    identity_key: IdentityKey,
+    role: NodeRole,
+    port: Option<u16>,
+}
+
+impl RefreshData {
+    pub fn new(host: String, identity_key: IdentityKey, role: NodeRole, port: Option<u16>) -> Self {
+        RefreshData {
+            host,
+            identity_key,
+            role,
+            port,
+        }
+    }
+
+    pub fn host(&self) -> String {
+        self.host.clone()
+    }
+
+    pub fn identity_key(&self) -> IdentityKey {
+        self.identity_key.clone()
+    }
+
+    pub fn port(&self) -> Option<u16> {
+        self.port
+    }
+
+    pub fn role(&self) -> NodeRole {
+        self.role.clone()
+    }
 }
 
 #[async_trait]
 impl CacheItemProvider for NodeDescriptionProvider {
     type Item = HashMap<IdentityKey, NymNodeDescription>;
     type Error = NodeDescribeCacheError;
 
-    async fn wait_until_ready(&self) {
-        self.contract_cache.wait_for_initial_values().await
```

Comment:
I'm worried about blocking here, do we know how long this might take, I'd rather return nothing until we have something, let the clients deal with empty stuff, then again if this is fast then we can just block, but I'm worried it won't be fast